modules/product/src/Entity/ProductVariation.php

```diff
@@ -500,4 +502,227 @@ public static function getCurrentUserId() {
     return [\Drupal::currentUser()->id()];
   }
 
+  /**
+   * Default value callback for 'sku' base field definition.
+   *
+   * @see ::baseFieldDefinitions()
+   *
+   * @return string
+   *   A prefixed unique identifier based on the current time in microseconds.
+   */
+  public static function getUniqSku() {
+    return \uniqid('sku-');
+  }
+
+  /**
+   * An AJAX callback to create all possible variations for commerce_product.
+   *
+   * @see commerce_product_field_widget_form_alter()
+   *
+   * @param array $form
+   *
+   *   An array form for commerce_product.
+   *
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *
+   *   The form state of the commerce_product form with at least one variation
+   *   created.
+   */
+  public static function createAllVariations(array $form, FormStateInterface $form_state) {
```

Comment:
Perhaps bulk variation creation should go into its own service? So it could be run without just IEF?